pydarn/plotting/fan.py

```diff
@@ -0,0 +1,248 @@
+
+        fan_shape = beam_corners_lons.shape
+
+        beam_corners_aacgm_lons = np.zeros((fan_shape[0], fan_shape[1]))
+        beam_corners_aacgm_lats = np.zeros((fan_shape[0], fan_shape[1]))
+        beam_corners_mlts = np.zeros((fan_shape[0], fan_shape[1]))
+
+        # Time for coordinate conversion
+        dtime = dt.datetime(dmap_data[plot_beams[0][0]]['time.yr'],
+                            dmap_data[plot_beams[0][0]]['time.mo'], dmap_data[plot_beams[0][0]]['time.dy'],
+                            dmap_data[plot_beams[0][0]]['time.hr'], dmap_data[plot_beams[0][0]]['time.mt'],
+                            dmap_data[plot_beams[0][0]]['time.sc'])
+        for x in range(fan_shape[0]):
+            for y in range(fan_shape[1]):
+                # Convert to AACGM
+                geomag = np.array(aacgmv2.
+                                  get_aacgm_coord(beam_corners_lats[x, y],
+                                                  beam_corners_lons[x, y],
+                                                  250, dtime))
+
+                beam_corners_aacgm_lats[x, y] = geomag[0]
+                beam_corners_aacgm_lons[x, y] = geomag[1]
+
+        # Work out shift due in MLT
+        mltshift = beam_corners_lons[x, y] - \
+            (aacgmv2.convert_mlt(beam_corners_lons[x, y], dtime) * 15)
+        beam_corners_mlts = beam_corners_aacgm_lons - mltshift
+
+        # Hold the beam positions
+        thetas = np.radians(beam_corners_mlts)
+        rs = beam_corners_aacgm_lats
+
+        # Get range-gate data and groundscatter array for given scan
+        scan = np.zeros((fan_shape[0] - 1, fan_shape[1]-1))
+        grndsct = np.zeros((fan_shape[0] - 1, fan_shape[1]-1))
+        iterat = 0
+        for i in np.nditer(plot_beams):
+            try:
+                slist = dmap_data[i.astype(int)]['slist']
+                scan[slist, iterat] = dmap_data[i.astype(int)][parameter]
+                grndsct[slist, iterat] = dmap_data[i.astype(int)]['gflg']
+                iterat += 1
+            except KeyError:
+                continue
+
+        # Colour table and max value selection depending on parameter plotted
+        # Load defaults if none given
+        # TODO: use cmaps as over writting cmap is bad practice...
+        # did I do that in my code ... hmm
+        if cmap is None:
+            cmap = {'p_l': 'plasma', 'v': PyDARNColormaps.PYDARN_VELOCITY,
+                    'w_l': PyDARNColormaps.PYDARN_VIRIDIS,
+                    'elv': PyDARNColormaps.PYDARN}
+            cmap = plt.cm.get_cmap(cmap[parameter])
+
+        # Setting zmin and zmax
+        defaultzminmax = {'p_l': [0, 50], 'v': [-200, 200],
+                          'w_l': [0, 250], 'elv': [0, 50]}
+        if zmin is None:
+            zmin = defaultzminmax[parameter][0]
+        if zmax is None:
+            zmax = defaultzminmax[parameter][1]
+
+        # Setup plot
+        # TODO: eek! remeber there is an option to pass in a variable called ax
```

Comment:
Did you see this TODO?

Reply:
Which one?

Not sure what to do in place of how cmap is implemented, as I'm very limited in Python knowledge right now